### PR TITLE
Task/bws56 group creation api

### PR DESCRIPTION
**Pull request by Donymak**

### CODE REVIEW

**Comment by Donymak** [comment]
image
Now it works this way

File: backend/src/data/repositories/group/group.repository.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    return this.#GroupModel
+      .query()
+      .select(`${TableName.GROUPS}`)
+      .join(`${TableName.USERS_GROUPS}`, 'group_id', `${TableName.GROUPS}.id`)
+      .join(`${TableName.WORKERS}`, 'id', `${TableName.USERS_GROUPS}.user_id`)
+      .select(`${TableName.GROUPS}.name`)
+      .where({ name: name })
+      .andWhere({ tenant_id: tenantId })
+      .select(`${TableName.WORKERS}.name`);
+  }
```
Comment (Donymak) [inline backend/src/data/repositories/group/group.repository.ts:47]: How to refactor this?

File: backend/src/data/repositories/group/group.repository.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      .andWhere({ tenantId })
+      .first();
+
+    if (!group) {
+      return null;
+    }
+
+    return Group.modelToEntity(group);
+  }
+  //
```
Comment (what1s1ove) [inline backend/src/data/repositories/group/group.repository.ts]:
```suggestion
```

File: shared/src/common/types/eam-group/eam-group-response-dto.type.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+type EAMGroupResponseDto = {
+  id: string;
+  name: string;
+  createdAt: Date;
+  tenantId: string;
+  // users: string[];
+  // permissions: string[];
```
Comment (what1s1ove) [inline shared/src/common/types/eam-group/eam-group-response-dto.type.ts]:
```suggestion
```

File: shared/src/common/enums/api/groups-api-path.enum.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+enum GroupsApiPath {
+  ROOT = '/group',
+  EDIT = '/group/edit/',
+}
```
Comment (what1s1ove) [inline shared/src/common/enums/api/groups-api-path.enum.ts]:
```suggestion
enum EAMApiPath {
  ROOT = '/'
  GROUPS = '/groups',
}
```
we should handle `edit` by `PUT` method

File: shared/src/common/enums/api/groups-api-path.enum.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+enum GroupsApiPath {
+  ROOT = '/',
+  EDIT = '/groups',
+}
```
Comment (what1s1ove) [inline shared/src/common/enums/api/groups-api-path.enum.ts]:
```suggestion
enum EAMApiPath {
  ROOT = '/',
  GROUPS = '/groups',
}
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/api/api.ts [hunk 6]
```diff
@@ -1,10 +1,11 @@
 import { FastifyPluginAsync } from 'fastify';
 import { ApiPath } from '~/common/enums/enums';
-import { master, token, tenant } from '~/services/services';
+import { master, token, tenant, group } from '~/services/services';
 import { initMastersApi } from './masters/masters.api';
 import { initTenantsApi } from './tenants/tenants.api';
 import { authorization as authorizationPlugin } from '~/plugins/plugins';
 import { WHITE_ROUTES } from '~/common/constants/constants';
+import { initGroupApi } from '~/api/groups/groups.api';
```
Comment (what1s1ove) [inline backend/src/api/api.ts]: should be part of eam api
```suggestion
import { initEAMApi } from '~/api/eam/eam.api';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/group/group.repository.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      .first();
+
+    if (!group) {
+      return null;
+    }
+
+    return Group.modelToEntity(group);
+  }
+
+  async create(group: GroupEntity): Promise<GroupM> {
```
Comment (Yaroslaww-1) [inline backend/src/data/repositories/group/group.repository.ts]: Repository should not return Model. It should return Entity

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/group/group.repository.ts (hunk 7 above)
Comment (Donymak) [reply backend/src/data/repositories/group/group.repository.ts:33]: But in the master repository create method returns also model:
image

Comment (Donymak) [reply backend/src/data/repositories/group/group.repository.ts]: > Repository should not return Model. It should return Entity

But in the master repository create method returns also model:
image

**Review by what1s1ove — APPROVED** [review]

**Review by Yaroslaww-1 — APPROVED** [review]

**Pull request merged**